analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/EmptyCollectionsShouldNotBeEnumeratedBase.cs

```diff
@@ -101,14 +101,28 @@ public abstract class EmptyCollectionsShouldNotBeEnumeratedBase : SymbolicRuleCh
         nameof(List<int>.AddRange),
         nameof(List<int>.Insert),
         nameof(List<int>.InsertRange),
-        nameof(HashSet<int>.SymmetricExceptWith),
         nameof(HashSet<int>.UnionWith),
+        nameof(HashSet<int>.SymmetricExceptWith),   // This can add and/or remove items => It should remove all CollectionConstraints.
+                                                    // However, ust learning NotEmpty (and thus unlearning Empty) is good enough for now.
```

Comment:
I added a checkbox for this in #7457 but this is super low priority.

Reply:
Sounds good to me.
  Please fix something that seems to be a typo in the second line of the comment.
  `However, just learning NotEmpty`.